src/cunumeric/matrix/batched_cholesky_template.inl

```diff
@@ -0,0 +1,130 @@
+/* Copyright 2021-2022 NVIDIA Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#pragma once
+
+// Useful for IDEs
+#include <core/task/exception.h>
+#include "cunumeric/cunumeric.h"
+#include "cunumeric/matrix/batched_cholesky.h"
+#include "cunumeric/matrix/potrf_template.inl"
+#include "cunumeric/matrix/transpose_template.inl"
+
+namespace cunumeric {
+
+using namespace legate;
+
+template <VariantKind KIND, Type::Code CODE>
+struct BatchedCholeskyImplBody {
+  template <class T>
+  void operator()(T* array, int32_t m, int32_t n)
+  {
+    PotrfImplBody<KIND, CODE>()(array, m, n);
+  }
+};
+
+template <VariantKind KIND, Type::Code CODE>
+struct BatchedTransposeImplBody {
+  using VAL = legate_type_of<CODE>;
+
+  void operator()(VAL* array, int32_t n);
+};
+
+template <Type::Code CODE>
+struct _cholesky_supported {
+  static constexpr bool value = CODE == Type::Code::FLOAT64 || CODE == Type::Code::FLOAT32;
+};
+
+template <VariantKind KIND>
+struct BatchedCholeskyImpl {
+  template <Type::Code CODE, int DIM>
+  void operator()(Array& input_array, Array& output_array) const
+  {
+    using VAL = legate_type_of<CODE>;
+
+    auto shape = input_array.shape<DIM>();
+    if (shape != output_array.shape<DIM>()) {
+      throw legate::TaskException(
+        "Batched cholesky is not yet supported when input/output types differ");
+    }
+
+    if (shape.empty()) return;
+
+    size_t strides[DIM];
+
+    auto input  = input_array.read_accessor<VAL, DIM>(shape).ptr(shape, strides);
+    auto output = output_array.write_accessor<VAL, DIM>(shape).ptr(shape, strides);
+
+    // TODO: we need some sort of check here on the strides
+    // This should be a dense thing.
```

Comment:
should strides just be == 1 ?

Reply:
they should. for adding a sanity check, you mean? or just removing the argument? this was also copied from potrf, although I really don't understand why it's done.

Reply:
I was just thinking that we could remove this comment and add check for strides.

Reply:
You can check the strides on the accessors to ensure that it's the way you want, see the matrix multiplication task implementations

Reply:
@jjwilke please follow on this comment, otherwise LGTM

Reply:
Yep, thanks for catching that. This did get added. Just forgot to remove the comment. Fixed.